packages/api/src/controllers/access-control.ts

```diff
@@ -50,73 +57,184 @@ app.post(
+          );
+        }
+
+        const query = [];
+        query.push(sql`signing_key.data->>'publicKey' = ${req.body.pub}`);
+        const [signingKeyOutput] = await db.signingKey.find(query, {
+          limit: 2,
+        });
+
+        if (signingKeyOutput.length == 0) {
+          console.log(`
+            access-control: gate: content with playbackId ${playbackId} is gated but corresponding public key not found for key ${req.body.pub}, disallowing playback
+          `);
+          throw new ForbiddenError(
+            "Content is gated and corresponding public key not found"
+          );
+        }
+
+        if (signingKeyOutput.length > 1) {
+          let collisionKeys = JSON.stringify(signingKeyOutput);
+          console.log(`
+            access-control: gate: content contentId ${content.id} with playbackId=${playbackId} is gated but multiple (${signingKeyOutput.length}) public keys found for key ${req.body.pub}, disallowing playback, colliding keys=${collisionKeys}
+          `);
+          throw new BadRequestError(
+            "Multiple signing keys found for the same public key."
+          );
+        }
+
+        const signingKey = signingKeyOutput[0];
+
+        if (signingKey.userId !== content.userId) {
+          console.log(`
+            access-control: gate: disallowing playback for contentId=${content.id} with playbackId=${playbackId} the content and the public key pub=${req.body.pub} do not share the same owner, signingKeyUserId=${signingKey.userId}, contentUserId=${content.userId}
+          `);
+          throw new NotFoundError("Content not found");
+        }
+
+        if (signingKey.disabled || signingKey.deleted) {
+          const signingKeyLog = JSON.stringify(JSON.stringify(signingKey));
+          console.log(`
+            access-control: gate: disallowing playback for contentId=${content.id} with playbackId=${playbackId} the public key pub=${signingKey.id} is disabled or deleted, signingKey=${signingKeyLog}
+          `);
+          throw new ForbiddenError("The public key is disabled or deleted");
+        }
+
+        tracking.recordSigningKeyValidation(signingKey.id);
+        res.set("Cache-Control", "max-age=120,stale-while-revalidate=600");
+        res.status(204);
+        return res.end();
+      case "webhook":
+        if (!req.body.accessKey) {
+          console.log(`
+            access-control: gate: no accessKey provided for playbackId ${playbackId}, disallowing playback
+          `);
+          throw new ForbiddenError(
+            "Content is gated and requires an access key"
+          );
+        }
+        const webhook = await db.webhook.get(content.playbackPolicy.webhookId);
+        if (!webhook) {
+          console.log(`
+            access-control: gate: content with playbackId ${playbackId} is gated but corresponding webhook not found for webhookId ${content.playbackPolicy.webhookId}, disallowing playback
+          `);
+          throw new ForbiddenError(
```

Comment:
Actually about status codes in the response I was going to use the status codes defined here 
https://www.notion.so/livepeer/Signing-Keys-for-Access-Control-8879f96a5aaf409eb4042acc281f5caa?pvs=4#6b07279a3e234715ad991b1e58dffb6e

So in this case it would be `403 Forbidden` because it's the same as when we don't find the SigningKey, but I am open to change both to InternalServerError or NotFound

Reply:
Not a strong opinion, feel free to keep 403! I didn't remember that original doc/decision.

Just to explain what I meant: This is an invalid state in our own database altogether which our code let happen! The user should never be able to create an object (asset) that references something (webhook) that does not exist. So it's not an user error but our own and hence the 500. The webhook might've been deleted, but in this case it'd have a `deleted: true` anyway and not not exist here completely.

Speaking of that, we might want to check the `deleted` field of the webhook here 👀 (on that case a 403 makes sense tho, so lets just stick to that)

Reply:
Sounds good! let's make it InternalServerError